packages/material-ui-lab/src/TreeView/TreeView.js

```diff
@@ -127,42 +122,52 @@ const TreeView = React.forwardRef(function TreeView(props, ref) {
     // try to get parent's next sibling
     const parent = nodeMap.current[node.parentId];
     if (parent) {
-      const parentSiblings = getChildren(parent.parentId);
+      const parentSiblings = getChildrenIds(parent.parentId);
       return parentSiblings[parentSiblings.indexOf(parent.id) + 1];
     }
     return null;
   };
 
   const getPreviousNode = (id) => {
     const node = nodeMap.current[id];
-    const siblings = getChildren(node.parentId);
+    const siblings = getChildrenIds(node.parentId);
     const nodeIndex = siblings.indexOf(id);
 
     if (nodeIndex === 0) {
       return node.parentId;
     }
 
-    // start with previous node
     let currentNode = siblings[nodeIndex - 1];
-    while (isExpanded(currentNode) && getChildren(currentNode).length > 0) {
-      currentNode = getChildren(currentNode).pop();
+    while (isExpanded(currentNode) && getChildrenIds(currentNode).length > 0) {
+      currentNode = getChildrenIds(currentNode).pop();
     }
 
     return currentNode;
   };
 
   const getLastNode = () => {
-    let lastNode = getChildren(null).pop();
+    let lastNode = getChildrenIds(null).pop();
 
     while (isExpanded(lastNode)) {
-      lastNode = getChildren(lastNode).pop();
+      lastNode = getChildrenIds(lastNode).pop();
     }
     return lastNode;
   };
-  const getFirstNode = () => getChildren(null)[0];
+  const getFirstNode = () => getChildrenIds(null)[0];
   const getParent = (id) => nodeMap.current[id].parentId;
 
-  const getOrder = (nodeAId, nodeBId) => {
+  /**
+   * This is used to determine the start and end of a selection range so
+   * we can get the nodes between the two border nodes.
+   *
+   * It finds the nodes' common ancestor using
+   * a naive implementation of a lowest common ancestor algorithm
+   * (https://en.wikipedia.org/wiki/Lowest_common_ancestor) and
+   * then compares the ancestors below it that are also above the
```

Comment:
Quite the sentence here :smile: So if I understand this correctly you mean the children of the LCA that are also above (siblings included?) "`nodeA` and `nodeB`" (or "`nodeA` or `nodeB`")?

Reply:
Correct, I'm going to look at the wording again it's quite hard to convey this nicely